photon-lib/src/main/scala/com/linkedin/photon/ml/hyperparameter/estimators/kernels/StationaryKernel.scala

```diff
@@ -56,6 +58,22 @@ abstract class StationaryKernel(
    */
   protected[kernels] def fromPairwiseDistances(dists: DenseMatrix[Double]): DenseMatrix[Double]
 
+  /**
+   * Unwraps hyper-parameters from scaled values to true values
+   *
+   * @param x the matrix of points, where each of the m rows is a point in the space and some columns contain scaled values
+   * @return the unwrapped x that all the columns have true values
+   */
+  protected[kernels] def unwrap(x: DenseMatrix[Double]): DenseMatrix[Double] = {
+    val u = x.copy
+    indexedTransformMap.map { case (index: Int, transform: String) => transform match {
+        case "LOG" => u(::, index) := u(::, index).map(element => Math.pow(10, element))
+        case "SQRT" => u(::, index) := u(::, index).map(element => Math.pow(element, 2))
```

Comment:
Let's pass the function directly (e.g. `Map(0 -> (x: Double) => Math.pow(10, x))`. It will make this code cleaner, and it will also remove the burden from keeping a list of transformations here -- the kernel doesn't really need to have this information.